transport/http/client.go

```diff
@@ -49,10 +50,37 @@ func (c ClientHandler) Handle(ctx context.Context, input interface{}) (
 
 	resp, err := c.client.Do(req.Build(ctx))
 	if err != nil {
-		return nil, metadata, err
+		err = &RequestSendError{Err: err}
+
+		// Override the error with a context canceled error, if that was canceled.
+		select {
+		case <-ctx.Done():
+			err = &smithy.CanceledError{Err: ctx.Err()}
+		default:
+		}
 	}
 
-	return &Response{Response: resp}, metadata, nil
+	return &Response{Response: resp}, metadata, err
+}
+
+// RequestSendError provides a generic request transport error.
+type RequestSendError struct {
```

Comment:
Can you add more documentation as to what type of errors should/will be wrapped by the `RequestSendError`? By the name, i would think they are connection errors